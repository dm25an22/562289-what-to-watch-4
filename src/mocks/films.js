
const COUNT_FILMS = 1;

const text = `Lorem Ipsum is simply dummy text of the printing and typesetting industry.
 Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer
  took a galley of type and scrambled it to make a type specimen book.`;

const films = [
  {
    title: `Fantastic Beasts`,
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Kids & Family`
  },
  {
    title: `Bohemian Rhapsody`,
    src: `img/bohemian-rhapsody.jpg`,
    genre: `Dramas`
  },
  {
    title: `Macbeth`,
    src: `img/macbeth.jpg`,
    genre: `Thrillers`
  },
  {
    title: `Revenant`,
    src: `img/revenant.jpg`,
    genre: `Thrillers`
  },
  {
    title: `Johnny English`,
    src: `img/johnny-english.jpg`,
    genre: `Kids & Family`
  },
  {
    title: `War of the Worlds`,
    src: `img/war-of-the-worlds.jpg`,
    genre: `Thrillers`
  },
  {
    title: `Seven Years in Tibet`,
    src: `img/seven-years-in-tibet.jpg`,
    genre: `Documentary`
  },
  {
    title: `Snatch`,
    src: `img/snatch.jpg`,
    genre: `Romance`
  },
  {
    title: `Mindhunter`,
    src: `img/mindhunter.jpg`,
    genre: `Horror`
  },
  {
    title: `Pulp Fiction`,
    src: `img/pulp-fiction.jpg`,
    genre: `Crime`
  },
  {
    title: `What we do in the shadows`,
    src: `img/what-we-do-in-the-shadows.jpg`,
    genre: `Sci-Fi`
  },
  {
    title: `Shutter Island`,
    src: `img/shutter-island.jpg`,
    genre: `Comedies`
  }
];

const actorsList = [
  `Robert Agnew`,
  `Jonathan Ahdout`,
  `Waris Ahluwalia`,
  `Ahmed Ahmed`,
  `Philip Ahn`,
  `Charles Aidman`,
  `Danny Aiello`,
  `Danny Aiello`,
  `Liam Aiken`,
  `Alan Aisenberg`,
  `Spottiswoode Aitken`,
  `Franklyn Ajaye`,
  `Ayad Akhtar`,
  `Gbenga Akinnagbe`,
  `Claude Akins`,
  `Marc Alaimo`,
  `Mark Alan`,
  `Matthew Alan`,
  `Rico Alaniz`,
  `Carlos Alazraqui`,
  `Luis Alberni`,
  `Devin Albert`,
  `Eddie Albert`,
];

const producers = [
  `Edward Janney`,
  `Sabra Jones`,
  `Tom Ruegger`,
  `Ron Senkowski`,
  `Kelly Leonard`,
  `Matthew Daniels`,
  `Olivia Luccardi`,
  `P. J. Wolfson`,
  `Yance Ford`,
  `Allen Estrin`,
  `Jenno Topping`,
];

const previews = [
  `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
];

const getRandomNumber = (min, max) => {
  min = Math.ceil(min);
  max = Math.floor(max);
  return Math.floor(Math.random() * (max - min + 1)) + min;
};

const getRandomRating = () => {
  const firstNumber = getRandomNumber(0, 10);
  return Number(`${firstNumber}.${firstNumber !== 10 && firstNumber !== 0 ? getRandomNumber(0, 9) : 0}`);
};

const getDescriptionRating = (rating) => {

  switch (true) {
    case rating < 3:
      return `Bad`;

    case rating >= 3 && rating < 5:
      return `Normal`;

    case rating >= 5 && rating < 8:
      return `Good`;

    case rating >= 8 && rating < 10:
      return `Very good`;

    case rating >= 10:
      return `Awesome`;
  }

  return `Unknown`;
};

const getRandomItem = (arr) => {
  return arr[getRandomNumber(0, arr.length - 1)];
};

const getRandomArrWithUniqeItems = (arr) => {
  const copyArr = [...arr];
  const result = [];
  for (let i = 0; i < getRandomNumber(2, arr.length - 1); i++) {
    result.push(copyArr.splice(getRandomNumber(0, copyArr.length - 1), 1)[0]);
  }

  return result;
};


const generateMock = () => {
  const ratingNumber = getRandomRating();
  const ratingString = getDescriptionRating(ratingNumber);
  const randomFilm = getRandomItem(films);

  return {
    title: randomFilm.title,
    moviePoster: `https://picsum.photos/300/450`,
    bigPoster: `https://picsum.photos/1900/550`,
    smallCardImg: randomFilm.src,
    genre: randomFilm.genre,
    year: getRandomNumber(1995, 2017),
    description: text,
    preview: getRandomItem(previews),
    rating: ratingNumber,
    descriptionRating: ratingString,
    quantityVotes: getRandomNumber(5, 300),
    producer: getRandomItem(producers),
    listActors: getRandomArrWithUniqeItems(actorsList)
  };
};

const createMock = (countFilms) => {
  return new Array(countFilms)
    .fill()
    .map(generateMock);
};

export default createMock(COUNT_FILMS);

