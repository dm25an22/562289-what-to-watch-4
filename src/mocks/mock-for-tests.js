const mockFilms = [
  {
    title: `Fantastic Beasts`,
    moviePoster: `img/the-grand-budapest-hotel-poster.jpg`,
    bigPoster: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    year: 2011,
    description: `Something`,
    rating: 4.7,
    descriptionRating: `Normal`,
    quantityVotes: 125,
    producer: `Tom Ruegger`,
    listActors: [`one`, `two`],
    smallCardImg: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    backgroundColor: `red`,
    runTime: 90,
    id: 1,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    title: `Title`,
    moviePoster: `img/the-grand-budapest-hotel-poster.jpg`,
    bigPoster: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    year: 2011,
    description: `Something`,
    rating: 4.7,
    descriptionRating: `Normal`,
    quantityVotes: 125,
    producer: `Tom Ruegger`,
    listActors: [`one`, `two`],
    smallCardImg: `img/bohemian-rhapsody.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    backgroundColor: `red`,
    runTime: 90,
    id: 2,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    title: `Title2`,
    moviePoster: `img/the-grand-budapest-hotel-poster.jpg`,
    bigPoster: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Action`,
    year: 2011,
    description: `Something`,
    rating: 4.7,
    descriptionRating: `Normal`,
    quantityVotes: 125,
    producer: `Tom Ruegger`,
    listActors: [`one`, `two`],
    smallCardImg: `img/macbeth.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    backgroundColor: `red`,
    runTime: 90,
    id: 3,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
];

const mockPromo = {
  title: `Promo`,
  moviePoster: `film.poster_image`,
  bigPoster: `film.background_image`,
  smallCardImg: `film.preview_image`,
  genre: `film.genre`,
  year: 1998,
  description: `film.description`,
  preview: `film.preview_video_link`,
  rating: 3.2,
  descriptionRating: `Bad`,
  quantityVotes: 1290,
  producer: `film.director`,
  listActors: [`1`, `2`, `3`],
  runTime: 120,
  videoLink: `film.video_link`,
  isFavorite: true,
  backgroundColor: `red`,
  id: 4,
};

const userDataMock = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `img/1.png`
};

const noop = () => {};

const commentsMock = [
  {
    "id": 1,
    "user": {
      "id": 4,
      "name": `Kate Muir`
    },
    "rating": 8.9,
    "comment": `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    "date": `2019-05-08T14:13:56.569Z`
  },
  {
    "id": 2,
    "user": {
      "id": 4,
      "name": `Kate Muir`
    },
    "rating": 2.0,
    "comment": `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    "date": `2019-05-08T14:13:56.569Z`
  },
  {
    "id": 3,
    "user": {
      "id": 4,
      "name": `Kate Muir`
    },
    "rating": 4.3,
    "comment": `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    "date": `2019-05-08T14:13:56.569Z`
  }
];

export {
  mockFilms,
  mockPromo,
  userDataMock,
  noop,
  commentsMock
};
