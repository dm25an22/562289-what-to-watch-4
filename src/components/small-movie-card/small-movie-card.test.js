import React from "react";
import renderer from "react-test-renderer";
import SmallMovieCard from "./small-movie-card.jsx";
import {mockFilms} from "../../mocks/mock-for-tests";
import {Router} from "react-router-dom";
import {history} from "../../history";


it(`Render SmallMovieCard`, () => {
  const film = mockFilms[1];

  const tree = renderer.create(
      <Router history={history}>
        <SmallMovieCard
          film={film}
          index={1}
          renderVideoPlayer={() => {}}
          startPlayHandler={() => {}}
          stopPlayHandler={() => {}}
          isPlaing={false}
        />
      </Router>, {
        createNodeMock: () => {
          return {};
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
