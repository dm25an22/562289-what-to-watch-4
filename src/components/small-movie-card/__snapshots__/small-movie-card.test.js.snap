// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render SmallMovieCard 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  />
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/film/2"
      onClick={[Function]}
    >
      Title
    </a>
  </h3>
</article>
`;
